### ISSUE TITLE
FZF feels less tolerant of typos (gif example)

### ISSUE THREAD

**Issue by pooriajr** [junegunn/fzf#3648]
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
- [x ] I have searched through the existing issues

## Info

- OS
    - [ ] Linux
    - [x] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [ ] zsh
    - [x ] fish

## Problem / Steps to reproduce

I don't know if this is a real issue or just a feeling, but I've noticed that FZF is feeling less tolerant lately. I used to be able to type very carelessly, lots of typos, and still always get to the file I was looking for. It felt like magic.

Suddenly for the past few weeks or so (I did a VimPlug update recently, don't know if that might have anything to do with it), I felt like if I made more than a single typo, it wouldn't find what I was looking for.

Today I narrowed it down to a case that shows what I mean. If I'm looking for a file called `search_icon.svg`, I can find it if I miss one single letter, such as `searc` or `serch`, but if I type out of order 'serach',  (a common typo for me), it won't find it.

Gif showing what I mean:
![CleanShot 2024-02-28 at 13 01 43](https://github.com/junegunn/fzf/assets/14811629/3ed799ff-7914-4b51-a085-7bfe23f8b40b)

Has it always been like this? And is there a way to make it more tolerant to this kind of typo?

Thanks so much for all your work on FZF btw!

**Comment by junegunn** [comment]
> Has it always been like this?

Yes, the algorithm of fzf isn't tolerant of out-of-order typos. When you type `abc`, you're looking for items that match the regular expression `a.*b.*c`, so the order of the characters is important. 

So here is my advice:

There's no need for a space between `sear...` and `ic`. The space splits the query into two terms and fzf looks for items that match `s.*e.*a.*r` **and** `i.*c.*`. This is useful in some cases (e.g. `jpg img` will match `image.jpg`), but in this particular case, you'll get better results by not splitting the query and typing something like `seic` (the scoring system of fzf is very good at acronym/prefix search). You type less, the intention is clearer (`s.*e.*i.*c`), and the scoring system of fzf works better in this way.

Does this answer your question?

**Comment by pooriajr** [comment]
Yes this helps! Thanks so much June, I'll start using it with less spaces as you suggested.

**Comment by diego898** [comment]
For reference, I made a discussion about this as well when I typed `wsudo` for example, results missed all of my sudo commands

@junegunn - is there any tweaks/options we can make locally to improve this behavior? Thanks!
